src/EditorFeatures/CSharpTest/GenerateFromMembers/AddConstructorParametersFromMembers/AddConstructorParametersFromMembersTests.cs

```diff
@@ -769,5 +779,177 @@ public C(int i, int j)
 }"
             );
         }
+
+        [WorkItem(33603, "https://github.com/dotnet/roslyn/issues/33603")]
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsAddConstructorParametersFromMembers)]
+        public async Task TestMultipleConstructors()
+        {
+            var source =
+@"
+class C
+{
+    int [|l|];
+    public C(int i)
+    {
+    }
+    public C(int i, int j)
+    {
+    }
+    public C(int i, int j, int k)
+    {
+    }
+}
+";
+            var expected1 =
+@"
+class C
+{
+    int l;
+    public C(int i, int l)
+    {
+        this.l = l;
+    }
+    public C(int i, int j)
+    {
+    }
+    public C(int i, int j, int k)
+    {
+    }
+}
+";
+            var expected2 =
+@"
+class C
+{
+    int l;
+    public C(int i)
+    {
+    }
+    public C(int i, int j, int l)
+    {
+        this.l = l;
+    }
+    public C(int i, int j, int k)
+    {
+    }
+}
+";
+            var expected3 =
+@"
+class C
+{
+    int l;
+    public C(int i)
+    {
+    }
+    public C(int i, int j, int l = 0)
+    {
+        this.l = l;
+    }
+    public C(int i, int j, int k)
+    {
+    }
+}
+";
+            await TestInRegularAndScriptAsync(source, expected1, 0);
+            await TestInRegularAndScriptAsync(source, expected2, 1);
+            await TestInRegularAndScriptAsync(source, expected3, 4);
```

Comment:
1. why are you not testing indices 2 and 3?
2. use a named parameter for the index parameter.
3. can you keep things as one-check per test method.  This approach here is painful in case changes happen.  you have to fix one case, tehn rerun, then fix the next, and so on.